### PR TITLE
AWS S3 setup and Snapshot save crud

### PR DESCRIPTION
**Pull request by somnath77**

### CODE REVIEW

File: app/controllers/api/v1/snapshots_controller.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        signer.presigned_url(:put_object,
+                             bucket: ENV['S3_BUCKET'],
+                             key: filename,
+                             acl: 'public-read',
+                             content_type: 'image/jpeg')
+      end
+
+      def find_drives_candidate
+        @drive = Drive.find(params[:drive_id])
+        @candidate = Candidate.find(params[:candidate_id])
```
Comment (rahulknojha) [inline app/controllers/api/v1/snapshots_controller.rb:46]: if we need this only drives_candidate, drive and candidate assignment is not required, and method is also not following single responsibility principle

File: config/routes.rb [hunk 2]
```diff
@@ -57,6 +57,8 @@
       end
 
       get '/drives/:drive_id/rules', to: 'rules#index'
+      resources :snapshots, only: %i[index create]
+      post 'presigned_url', to: 'snapshots#presigned_url'
```
Comment (rahulknojha) [inline config/routes.rb:61]: can be done with line no 60 with do... end block

**Comment by rahulknojha** [comment]
test cases missing

File: app/controllers/api/v1/snapshots_controller.rb (hunk 1 above)
Comment (somnath77) [reply app/controllers/api/v1/snapshots_controller.rb]: Resolved

File: config/routes.rb (hunk 2 above)
Comment (somnath77) [reply config/routes.rb]: Done

**Review by rahulknojha — APPROVED** [review]

**Pull request merged**